docs/v3/core/index.rst

```diff
@@ -1598,38 +1520,125 @@ Storage transformers may be stacked to combine different functionalities:
+array       data type               `array-metadata-data-type`_              :ref:`data-types-list`
+array       chunk grid              `array-metadata-chunk-grid`_             :ref:`chunk-grid-list`
+array       chunk key encoding      `array-metadata-chunk-key-encoding`_     :ref:`chunk-key-encoding-list`
+array       codecs                  `array-metadata-codecs`_                 :ref:`codecs-list`
+array       storage transformer     `array-metadata-storage-transformers`_   :ref:`storage-transformers-list`
+=========== ======================= ====================================== ================================
 
 If such extension points are used by groups or arrays, they are required.
 
-See https://github.com/zarr-developers/zarr-specs/issues/49 for a list of
-potential extensions.
+New extension points may be proposed to the Zarr community through the ZEP
+process. See `ZEP 0 <https://zarr.dev/zeps/active/ZEP0000.html>`_ for more information.
+
+Extension definition
+--------------------
+
+Extensions are defined in `metadata documents`_ either as objects or as
+short-hand names. If using an objection definition, the following pattern
+MUST be followed::
+
+    {
+        "name": "<name>",
+        "configuration": { ... } # optional
+    }
+
+If such an object is present, the field `must_understand` is implicitly set to
+`True` and an object can explicitly set `must_understand=False` if
+implementations can ignore its presence, following the current guidelines in
+the v3 specification.
+
+Instead of extension objects, short-hand names may continue to be used if no
+configuration metadata is required. They would be equivalent to extension
+objects with just a `name` key.
 
-Specifications for new extensions are recommended to be published in the
-https://github.com/zarr-developers/zarr-specs repository via the
-`ZEP process <https://zarr.dev/zeps/active/ZEP0000.html>`_. If a specification
-is published decentralized (e.g. for initial experimentation or due to a very
-specialized scope), it must use a URL in the `name` key of its metadata, which
-identifies the publishing organization or individual, and should point to the
-specification of the extension.
+Extension naming
+----------------
+
+The identifier used in the `name` field of the extension definition can follow one of two forms:
+
+1. **Raw names** MUST be assigned within a central repository and follow the
+   compatibility and versioning v3 `stability policy`_.
+   The name assignment is managed through the `zarr-extensions`_
+   Github repository, where each extension is
+   listed and either contains a spec document or links to a spec document.
+   Names are never unassigned or reassigned. The Zarr Steering Council or by delegation a
+   maintainer team reserves the right to refuse name assignment at its own
+   discretion.
+
+   - **Example:** ``zstd``
+   - **Acceptd regex:** ``^[a-z0-9-_.]+$``
```

Comment:
```suggestion
   - **Acceptd regex:** ``^[a-z0-9-_.]+$``
```
Maybe I'm alone here but I don't find regex useful. If that regex can be translated to english, this would be better.

Reply:
I think regexes are nice because they are unambiguous.